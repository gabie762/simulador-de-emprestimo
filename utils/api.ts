// utils/api.ts
import { Produto, Resultado } from '../types';

// Configuração da API
const API_BASE_URL = 'https://sua-api.com/api'; // Substitua pela URL da sua API

// Tipos para requests/responses
export interface CadastrarProdutoRequest {
  nome: string;
  taxaJurosAnual: number;
  prazoMaximo: number;
}

export interface SimulacaoRequest {
  produtoId: string;
  valor: number;
  prazo: number;
}

export interface SimulacaoResponse {
  produto: Produto;
  valorEmprestimo: number;
  prazo: number;
  taxaJurosAnual: number;
  taxaJurosMensal: number;
  valorParcela: number;
  totalPagar: number;
  totalJuros: number;
  memoriaCalculo?: {
    mes: number;
    parcela: number;
    juros: number;
    amortizacao: number;
    saldoDevedor: number;
  }[];
}

// Função helper para fazer requests
async function apiRequest<T>(
  endpoint: string,
  options: RequestInit = {}
): Promise<T> {
  const url = `${API_BASE_URL}${endpoint}`;
  
  const defaultHeaders = {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    // Adicione headers de autenticação se necessário
    // 'Authorization': `Bearer ${token}`,
  };

  const config: RequestInit = {
    ...options,
    headers: {
      ...defaultHeaders,
      ...options.headers,
    },
  };

  try {
    const response = await fetch(url, config);
    
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }
    
    const data = await response.json();
    return data;
  } catch (error) {
    console.error(`Erro na API (${endpoint}):`, error);
    throw error;
  }
}

// GET /produtos – Listar produtos
export async function getProdutos(): Promise<Produto[]> {
  try {
    const produtos = await apiRequest<Produto[]>('/produtos', {
      method: 'GET',
    });
    return produtos;
  } catch (error) {
    console.error('Erro ao buscar produtos:', error);
    throw new Error('Não foi possível carregar os produtos do servidor.');
  }
}

// POST /produtos – Cadastrar produto
export async function postProduto(produto: CadastrarProdutoRequest): Promise<Produto> {
  try {
    const novoProduto = await apiRequest<Produto>('/produtos', {
      method: 'POST',
      body: JSON.stringify(produto),
    });
    return novoProduto;
  } catch (error) {
    console.error('Erro ao cadastrar produto:', error);
    
    // Tratamento de erros específicos
    if (error instanceof Error) {
      if (error.message.includes('409')) {
        throw new Error('PRODUTO_DUPLICADO');
      }
      if (error.message.includes('400')) {
        throw new Error('Dados do produto inválidos.');
      }
    }
    
    throw new Error('Não foi possível cadastrar o produto.');
  }
}

// POST /simulações – Realizar simulação de empréstimo
export async function postSimulacao(simulacao: SimulacaoRequest): Promise<Resultado> {
  try {
    const resultado = await apiRequest<SimulacaoResponse>('/simulacoes', {
      method: 'POST',
      body: JSON.stringify(simulacao),
    });
    
    // Converte o response da API para o formato esperado pelo app
    const resultadoFormatado: Resultado = {
      produto: resultado.produto.nome,
      valorEmprestimo: resultado.valorEmprestimo.toFixed(2),
      prazo: resultado.prazo,
      taxaJurosAnual: resultado.taxaJurosAnual.toFixed(2) + '%',
      taxaJurosMensal: resultado.taxaJurosMensal.toFixed(2) + '%',
      valorParcela: resultado.valorParcela.toFixed(2),
      totalPagar: resultado.totalPagar.toFixed(2),
      totalJuros: resultado.totalJuros.toFixed(2),
      memoriaCalculo: resultado.memoriaCalculo?.map(item => ({
        mes: item.mes,
        parcela: item.parcela.toFixed(2),
        juros: item.juros.toFixed(2),
        amortizacao: item.amortizacao.toFixed(2),
        saldoDevedor: item.saldoDevedor.toFixed(2),
      })) || [],
    };
    
    return resultadoFormatado;
  } catch (error) {
    console.error('Erro ao realizar simulação:', error);
    
    // Tratamento de erros específicos
    if (error instanceof Error) {
      if (error.message.includes('400')) {
        throw new Error('Dados da simulação inválidos.');
      }
      if (error.message.includes('404')) {
        throw new Error('Produto não encontrado.');
      }
    }
    
    throw new Error('Não foi possível realizar a simulação.');
  }
}

// Função para deletar produto (se a API suportar)
export async function deleteProduto(id: string): Promise<boolean> {
  try {
    await apiRequest(`/produtos/${id}`, {
      method: 'DELETE',
    });
    return true;
  } catch (error) {
    console.error('Erro ao deletar produto:', error);
    throw new Error('Não foi possível deletar o produto.');
  }
}

// Função para testar conectividade com a API
export async function testApiConnection(): Promise<boolean> {
  try {
    // Tenta fazer uma requisição simples para verificar se a API está acessível
    await apiRequest('/produtos', { method: 'GET' });
    return true;
  } catch (error) {
    console.error('API não está acessível:', error);
    return false;
  }
}
