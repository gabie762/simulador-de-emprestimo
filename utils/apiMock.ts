// utils/apiMock.ts
import AsyncStorage from '@react-native-async-storage/async-storage';

const STORAGE_KEY = 'produtos_mock';
const defaultProdutos = [
  { id: '1', nome: 'Crédito Pessoal', taxaJurosAnual: 10.5, prazoMaximo: 24 },
  { id: '2', nome: 'Crediário CAIXA', taxaJurosAnual: 8.2, prazoMaximo: 36 },
  { id: '3', nome: 'Empréstimo Consignado', taxaJurosAnual: 6.8, prazoMaximo: 84 },
];

export async function getProdutos() {
  try {
    const stored = await AsyncStorage.getItem(STORAGE_KEY);
    if (stored) {
      return JSON.parse(stored);
    } else {
      await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(defaultProdutos));
      return [...defaultProdutos];
    }
  } catch (error) {
    console.error('Erro ao carregar produtos:', error);
    return [...defaultProdutos];
  }
}

export async function postProduto(produto: any) {
  return new Promise((resolve, reject) => {
    setTimeout(async () => {
      try {
        const stored = await AsyncStorage.getItem(STORAGE_KEY);
        const produtos = stored ? JSON.parse(stored) : [...defaultProdutos];
        
        // Verifica se já existe um produto com o mesmo nome
        const produtoExistente = produtos.find((p: any) => 
          p.nome.toLowerCase().trim() === produto.nome.toLowerCase().trim()
        );
        
        if (produtoExistente) {
          reject(new Error('PRODUTO_DUPLICADO'));
          return;
        }
        
        const novo = { ...produto, id: Date.now().toString() };
        produtos.push(novo);
        await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(produtos));
        resolve(novo);
      } catch (error) {
        console.error('Erro ao salvar produto:', error);
        reject(error instanceof Error ? error : new Error('Erro desconhecido'));
      }
    }, 500);
  });
}

export async function resetProdutos() {
  await AsyncStorage.removeItem(STORAGE_KEY);
  await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(defaultProdutos));
}

export async function deleteProduto(id: string) {
  try {
    const stored = await AsyncStorage.getItem(STORAGE_KEY);
    const produtos = stored ? JSON.parse(stored) : [];
    
    const produtosFiltrados = produtos.filter((produto: any) => produto.id !== id);
    await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(produtosFiltrados));
    
    return true;
  } catch (error) {
    console.error('Erro ao deletar produto:', error);
    return false;
  }
}
