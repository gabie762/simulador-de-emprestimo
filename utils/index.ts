import { Produto, Resultado } from '../types';
import { Platform } from 'react-native';

export const mockProdutos: Produto[] = [
  { id: '1', nome: 'Crédito Pessoal Padrão', taxaJurosAnual: 25.0, prazoMaximo: 48 },
  { id: '2', nome: 'Crédito Imobiliário', taxaJurosAnual: 9.5, prazoMaximo: 360 },
  { id: '3', nome: 'Financiamento Veicular', taxaJurosAnual: 18.2, prazoMaximo: 60 },
];

// Helper para normalizar estilos shadow para web
export const normalizeShadow = (shadowStyle: any) => {
  if (Platform.OS === 'web') {
    const { shadowColor, shadowOffset, shadowOpacity, shadowRadius, ...rest } = shadowStyle;
    
    if (shadowColor && shadowOffset && shadowOpacity !== undefined && shadowRadius) {
      return {
        ...rest,
        boxShadow: `${shadowOffset.width}px ${shadowOffset.height}px ${shadowRadius}px rgba(0, 0, 0, ${shadowOpacity})`,
      };
    }
  }
  
  return shadowStyle;
};

export const calcularSimulacao = (
  valorEmprestimo: number,
  prazo: number,
  produto: Produto
): Resultado => {
  const taxaMensal = (produto.taxaJurosAnual / 12) / 100;
  const parcela = valorEmprestimo * (taxaMensal * Math.pow(1 + taxaMensal, prazo)) / (Math.pow(1 + taxaMensal, prazo) - 1);
  const total = parcela * prazo;
  const totalJuros = total - valorEmprestimo;
  
  let saldoDevedor = valorEmprestimo;
  const memoriaCalculo = [];
  
  for (let i = 1; i <= prazo; i++) {
    const jurosMes = saldoDevedor * taxaMensal;
    const amortizacao = parcela - jurosMes;
    saldoDevedor -= amortizacao;
    memoriaCalculo.push({ 
      mes: i, 
      parcela: parcela.toFixed(2), 
      juros: jurosMes.toFixed(2), 
      amortizacao: amortizacao.toFixed(2), 
      saldoDevedor: Math.abs(saldoDevedor).toFixed(2) 
    });
  }

  return {
    produto: produto.nome,
    valorEmprestimo: valorEmprestimo.toFixed(2),
    prazo: prazo,
    taxaJurosAnual: produto.taxaJurosAnual.toFixed(2),
    taxaJurosMensal: (taxaMensal * 100).toFixed(4),
    valorParcela: parcela.toFixed(2),
    totalPagar: total.toFixed(2),
    totalJuros: totalJuros.toFixed(2),
    memoriaCalculo,
  };
};
