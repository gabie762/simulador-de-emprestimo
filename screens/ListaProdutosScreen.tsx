import React from 'react';
import { FlatList, Text, TouchableOpacity, View } from 'react-native';
import { globalStyles, productStyles } from '../styles';
import { ListaProdutosProps, Produto } from '../types';
import { deleteProduto, resetProdutos } from '../utils/apiMock';

interface ExtendedListaProdutosProps extends ListaProdutosProps {
  onProdutoDeleted?: () => void;
  testID?: string; // Added testID prop
}

const ListaProdutosScreen: React.FC<ExtendedListaProdutosProps> = ({ produtos, goHome, onProdutoDeleted, testID }) => {
  const handleDeleteProduto = async (produto: Produto) => {
    try {
      const resultado = await deleteProduto(produto.id);
      
      if (resultado) {
        if (onProdutoDeleted) {
          onProdutoDeleted();
        }
      }
    } catch (error) {
      console.error('Erro na exclusão:', error);
    }
  };

  const handleRestaurarProdutosPadrao = async () => {
    try {
      await resetProdutos();
      
      if (onProdutoDeleted) {
        onProdutoDeleted();
      }
    } catch (error) {
      console.error('Erro ao restaurar produtos padrão:', error);
    }
  };

  const renderProdutoItem = ({ item }: { item: Produto }) => (
    <View style={productStyles.produtoCard}>
      <View style={{ flexDirection: 'row', justifyContent: 'space-between', alignItems: 'flex-start' }}>
        <View style={{ flex: 1 }}>
          <Text style={productStyles.cardTitle}>{item.nome}</Text>
          <View style={productStyles.cardInfoContainer}>
            <Text style={productStyles.cardInfo}>
              <Text style={productStyles.cardLabel}>Taxa Anual:</Text> {item.taxaJurosAnual.toFixed(2)}%
            </Text>
            <Text style={productStyles.cardInfo}>
              <Text style={productStyles.cardLabel}>Prazo Máximo:</Text> {item.prazoMaximo} meses
            </Text>
          </View>
        </View>
        <TouchableOpacity 
          style={{
            backgroundColor: '#FF4444',
            paddingHorizontal: 12,
            paddingVertical: 8,
            borderRadius: 6,
            marginLeft: 10,
          }}
          onPress={() => handleDeleteProduto(item)}
        >
          <Text style={{ color: '#fff', fontSize: 12, fontWeight: 'bold' }}>Excluir</Text>
        </TouchableOpacity>
      </View>
    </View>
  );

  return (
    <View style={globalStyles.container} testID={testID}> {/* Applied testID */}
      <Text style={globalStyles.title}>Produtos Cadastrados</Text>
      
      {produtos.length === 0 ? (
        <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', paddingHorizontal: 20 }}>
          <Text style={{ textAlign: 'center', color: '#666', fontSize: 16, marginBottom: 20 }}>
            Nenhum produto cadastrado.
          </Text>
          <TouchableOpacity 
            style={[globalStyles.button, { backgroundColor: '#F7941E' }]} 
            onPress={handleRestaurarProdutosPadrao}
            testID="restaurar-produtos-button"
          >
            <Text style={globalStyles.buttonText}>Restaurar Produtos Padrão</Text>
          </TouchableOpacity>
        </View>
      ) : (
        <>
          <FlatList 
            data={produtos} 
            renderItem={renderProdutoItem} 
            keyExtractor={(item) => item.id} 
            style={globalStyles.fullWidthList}
          />
          
          <View style={{ marginTop: 20, gap: 10 }}>
            <TouchableOpacity 
              style={[globalStyles.button, { backgroundColor: '#F7941E' }]} 
              onPress={handleRestaurarProdutosPadrao}
            >
              <Text style={globalStyles.buttonText}>Restaurar Produtos Padrão</Text>
            </TouchableOpacity>
            
            <TouchableOpacity 
              style={[globalStyles.button, globalStyles.buttonSecondary]} 
              onPress={goHome}
              testID="voltar-button"
            >
              <Text style={globalStyles.buttonTextSecondary}>Voltar</Text>
            </TouchableOpacity>
          </View>
        </>
      )}
    </View>
  );
};

export default ListaProdutosScreen;
