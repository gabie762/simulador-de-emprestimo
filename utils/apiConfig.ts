// utils/apiConfig.ts
import * as ApiReal from './api';
import * as ApiMock from './apiMock';

// Configuração para alternar entre Mock e API real
export const USE_REAL_API = false; // Mude para true quando quiser usar a API real

// Exporta as funções baseado na configuração
export const getProdutos = USE_REAL_API ? ApiReal.getProdutos : ApiMock.getProdutos;
export const postProduto = USE_REAL_API ? ApiReal.postProduto : ApiMock.postProduto;
export const deleteProduto = USE_REAL_API ? ApiReal.deleteProduto : ApiMock.deleteProduto;

// Função de simulação (só existe na API real, no mock usamos o cálculo local)
export const postSimulacao = USE_REAL_API ? ApiReal.postSimulacao : null;

// Função para testar conectividade (só para API real)
export const testApiConnection = USE_REAL_API ? ApiReal.testApiConnection : null;
