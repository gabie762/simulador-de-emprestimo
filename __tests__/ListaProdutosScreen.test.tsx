import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import ListaProdutosScreen from '../screens/ListaProdutosScreen';
import { deleteProduto, resetProdutos } from '../utils/apiMock';

jest.mock('../utils/apiMock', () => ({
  deleteProduto: jest.fn(),
  resetProdutos: jest.fn(),
}));

describe('ListaProdutosScreen', () => {
  const mockProdutos = [
    { id: '1', nome: 'Produto 1', taxaJurosAnual: 5.5, prazoMaximo: 12 },
    { id: '2', nome: 'Produto 2', taxaJurosAnual: 6.0, prazoMaximo: 24 },
  ];

  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation(() => {});
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('renders correctly', () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();
    const { toJSON } = render(
      <ListaProdutosScreen produtos={[]} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders the empty state when no products are available', () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();
    const { getByText } = render(
      <ListaProdutosScreen produtos={[]} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );

    expect(getByText('Nenhum produto cadastrado.')).toBeTruthy();
    expect(getByText('Restaurar Produtos Padrão')).toBeTruthy();
  });

  it('renders the product list when products are available', () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();
    const { getByText } = render(
      <ListaProdutosScreen produtos={mockProdutos} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );

    expect(getByText('Produto 1')).toBeTruthy();
    expect(getByText('Produto 2')).toBeTruthy();
  });

  it('calls deleteProduto when the Excluir button is pressed', async () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();
    (deleteProduto as jest.Mock).mockResolvedValueOnce(true);

    const { getAllByText } = render(
      <ListaProdutosScreen produtos={mockProdutos} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );

    const deleteButtons = getAllByText('Excluir');
    fireEvent.press(deleteButtons[0]);

    await waitFor(() => {
      expect(deleteProduto).toHaveBeenCalledWith('1');
      expect(mockOnProdutoDeleted).toHaveBeenCalled();
    });
  });

  it('does not call onProdutoDeleted if deleteProduto returns false', async () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();
    (deleteProduto as jest.Mock).mockResolvedValueOnce(false);

    const { getAllByText } = render(
      <ListaProdutosScreen produtos={mockProdutos} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );

    const deleteButtons = getAllByText('Excluir');
    fireEvent.press(deleteButtons[0]);

    await waitFor(() => {
      expect(deleteProduto).toHaveBeenCalledWith('1');
      expect(mockOnProdutoDeleted).not.toHaveBeenCalled();
    });
  });

  it('handles errors when deleteProduto fails', async () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();
    (deleteProduto as jest.Mock).mockRejectedValueOnce(new Error('Delete failed'));

    const { getAllByText } = render(
      <ListaProdutosScreen produtos={mockProdutos} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );

    const deleteButtons = getAllByText('Excluir');
    fireEvent.press(deleteButtons[0]);

    await waitFor(() => {
      expect(deleteProduto).toHaveBeenCalledWith('1');
      expect(mockOnProdutoDeleted).not.toHaveBeenCalled();
    });
  });

  it('does not call onProdutoDeleted if it is undefined', async () => {
    const mockGoHome = jest.fn();
    (deleteProduto as jest.Mock).mockResolvedValueOnce(true);

    const { getAllByText } = render(
      <ListaProdutosScreen produtos={mockProdutos} goHome={mockGoHome} />
    );

    const deleteButtons = getAllByText('Excluir');
    fireEvent.press(deleteButtons[0]);

    await waitFor(() => {
      expect(deleteProduto).toHaveBeenCalledWith('1');
    });
  });

  it('calls resetProdutos when the Restaurar Produtos Padrão button is pressed', async () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();
    (resetProdutos as jest.Mock).mockResolvedValueOnce(undefined);

    const { getByText } = render(
      <ListaProdutosScreen produtos={[]} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );

    fireEvent.press(getByText('Restaurar Produtos Padrão'));

    await waitFor(() => {
      expect(resetProdutos).toHaveBeenCalled();
      expect(mockOnProdutoDeleted).toHaveBeenCalled();
    });
  });

  it('does not call onProdutoDeleted if resetProdutos fails', async () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();
    (resetProdutos as jest.Mock).mockRejectedValueOnce(new Error('Reset failed'));

    const { getByText } = render(
      <ListaProdutosScreen produtos={[]} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );

    fireEvent.press(getByText('Restaurar Produtos Padrão'));

    await waitFor(() => {
      expect(resetProdutos).toHaveBeenCalled();
      expect(mockOnProdutoDeleted).not.toHaveBeenCalled();
    });
  });

  it('does not call onProdutoDeleted if it is undefined during resetProdutos', async () => {
    const mockGoHome = jest.fn();
    (resetProdutos as jest.Mock).mockResolvedValueOnce(undefined);

    const { getByText } = render(
      <ListaProdutosScreen produtos={[]} goHome={mockGoHome} />
    );

    fireEvent.press(getByText('Restaurar Produtos Padrão'));

    await waitFor(() => {
      expect(resetProdutos).toHaveBeenCalled();
    });
  });

  it('calls goHome when the Voltar button is pressed', () => {
    const mockGoHome = jest.fn();
    const mockOnProdutoDeleted = jest.fn();

    const { getByText } = render(
      <ListaProdutosScreen produtos={mockProdutos} goHome={mockGoHome} onProdutoDeleted={mockOnProdutoDeleted} />
    );

    fireEvent.press(getByText('Voltar'));

    expect(mockGoHome).toHaveBeenCalled();
  });
});
