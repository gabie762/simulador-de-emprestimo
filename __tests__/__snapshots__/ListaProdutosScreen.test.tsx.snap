// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListaProdutosScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "flex-start",
      "maxWidth": 600,
      "padding": 20,
      "width": "100%",
    }
  }
>
   
  <Text
    style={
      {
        "color": "#0066B3",
        "fontFamily": "Caixa",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginBottom": 20,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Produtos Cadastrados
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 16,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Nenhum produto cadastrado.
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F7941E",
          "borderRadius": 8,
          "marginBottom": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="restaurar-produtos-button"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Caixa",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Restaurar Produtos Padr√£o
      </Text>
    </View>
  </View>
</View>
`;
